src/UNL/Knowledge/Records.php

```diff
@@ -124,4 +135,108 @@ protected function getMergedKey($item, $section, $tags)
 
     	return implode(' ', $results);
     }
+
+    protected function getPublicProperties()
+    {
+        $self = $this;
+        $getPublicProperties = function() use ($self) {
+            return get_object_vars($self);
+        };
+        $getPublicProperties = $getPublicProperties->bindTo(null, null);
+
+        return $getPublicProperties();
```

Comment:
I'm not a fan of ever using Reflection, but you are correct that this looks very hacky without some sort of comment to say what it does. We just need a map of object field to value.